Drop privileged securityContext for buildah steps 🏟

vdemeester opened the pull request:
This also set `STORAGE_DRIVER` to `vfs` to make sure we can run it in
a non privileged container.

Part of SRVKP-950

/cc @pradeepitm12 @piyush-garg @gabemontero @nikhil-thomas 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

gabemontero commented:
> This also set `STORAGE_DRIVER` to `vfs` to make sure we can run it in
> a non privileged container.
> 
> Part of SRVKP-950

thanks for beating me to the punch @vdemeester :-)

> 
> /cc @pradeepitm12 @piyush-garg @gabemontero @nikhil-thomas
> 
> Signed-off-by: Vincent Demeester [vdemeest@redhat.com](mailto:vdemeest@redhat.com)

gabemontero reviewed (commented):
@vdemeester so at least in my testing `buildah push` still needs

```
 securityContext:
      runAsUser: 0
```

we are fine with not setting `securityContext` with `buildah bud`

gabemontero commented:
Oh, and as a result, the associated READMEs @vdemeester need to explain how to add the `anyuid` SCC to the associated SA in order for `runAsUser: 0` to be allowed by OCP

vdemeester commented:
/hold

sbose78 commented:
>  add the anyuid SCC to the associated SA in order for runAsUser: 0 to be allowed by OCP

Hi @vdemeester 👋 
Are we waiting to have the `pipeline` SA configured to have the `anyuid` SCC ?

vdemeester commented:
> Are we waiting to have the `pipeline` SA configured to have the `anyuid` SCC ?

Waiting or wanting ? Indeed, we should have `pipeline` SA to have `anyuid` SCC for being able to run those.

sbose78 commented:
> Waiting or wanting

Haha, both.

On that note, I think both the `pipeline` sa and the controller `sa` should have `anyuid`. 

Starting with the former will help us be more explicit with what the `pipeline` service account needs to be able to do with the s2i / `buildah push`.

sbose78 commented:
> I think both the pipeline sa and the controller sa should have anyuid.

pipeline sa now has anyuid

vdemeester commented:
@sbose78 @gabemontero Updated
/cc @piyush-garg

vdemeester commented:
@gabemontero tested this real quick, I don't think we need the `runAsUser: 0` for push (without it it still works in my tests)

vdemeester committed:
Drop privileged securityContext for buildah steps 🏟

This also set `STORAGE_DRIVER` to `vfs` to make sure we can run it in
a non privileged container.

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

gabemontero commented:
> @gabemontero tested this real quick, I don't think we need the `runAsUser: 0` for push (without it it still works in my tests)

Hopefully  I can retry my test case from before soon @vdemeester and we can compare notes.

Among other things though `quay.io/buildah/stable:v1.11.0` is a bit old ... in fact I saw you using a later version in the upstream catalog PR.

I think we should update to 1.15 or 1.16

vdemeester commented:
@gabemontero an update for this has been merged very recently :stuck_out_tongue: I just need to rebase.

gabemontero commented:
OK @vdemeester I *almost* always saw us being able to remove `runAsUser: 0` but not quite ... the buildah push can still fail, even with v1.16.0 of the image, with:

```
[build] error creating build container: Error committing the finished image: error adding layer with blob "sha256:2b63304f1869b42027cc113af46692a963e05ef7867c88d36ef9c75ffc2fdfc7": Error processing tar file(exit status 1): there might not be enough IDs available in the namespace (requested 192:192 for /run/systemd/netif): lchown /run/systemd/netif: invalid argument
[build] level=error msg="exit status 125"

[push] 2020/11/30 19:02:45 Skipping step because a previous step failed

container step-push has failed  : [{"key":"StartedAt","value":"2020-11-30T19:02:45.318Z","type":"InternalTektonResult"}]
```

Specifics of my run:

- if I remove `runAsUser: 0` for the push in my task, but my serviceaccount *STILL* has the anyuid SCC, it will work
- however, if I remove anyuid from the serviceaccount, it fails with the error above 

Now, certainly if  our examples always cite the `pipeline` SA, which now has the anyuid SCC associated with it when pipelines are installed, we assume it stays that way, we only officially support that way, etc. etc. then we are OK.

However, IMO, we should minimally explain that anyuid is needed on the SA, in case folks try these tasks with other SA's besides the pipelines SA

WDYT ?

vdemeester commented:
> However, IMO, we should minimally explain that anyuid is needed on the SA, in case folks try these tasks with other SA's besides the pipelines SA
> 
> WDYT ?

Yes, I definitely agree on this. Main question is where should we document that ? I am not sure if there is a `ClusterTask` part on the OpenShift Pipeline documentation (cc @Preeticp ).

piyush-garg commented on task/buildah-pr/0.1/buildah-pr.yaml:
```diff
@@ -51,19 +51,18 @@ spec:
+      - name: STORAGE_DRIVER
+        value: vfs
 
-  volumes:
```
Should we remove this?

gabemontero replied:
no we still need this

vdemeester replied:
oh way, what did I remove that :sweat: will fix

vdemeester replied:
fixed

gabemontero commented:
other than the concern on docs (i.e. you need to have `anyuid` SCC on the SA for the `TaskRun`) I'm good here

presumably it is easily understood that `buildah` consumes the `STORAGE_DRIVER` env

chmouel commented:
/lgtm

vdemeester commented:
/hold cancel

gabemontero commented:
@vdemeester @chmouel - were you all ever able to link up with @Preeticp about having the docs cover https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-735987554 re: https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-736252584

gabemontero commented:
> @vdemeester @chmouel - were you all ever able to link up with @Preeticp about having the docs cover [#56 (comment)](https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-735987554) re: [#56 (comment)](https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-736252584)

I found the doc item https://issues.redhat.com/browse/RHDEVDOCS-2384

I'll post some comments there

gabemontero commented:
> > @vdemeester @chmouel - were you all ever able to link up with @Preeticp about having the docs cover [#56 (comment)](https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-735987554) re: [#56 (comment)](https://github.com/openshift/pipelines-catalog/pull/56#issuecomment-736252584)
> 
> I found the doc item https://issues.redhat.com/browse/RHDEVDOCS-2384
> 
> I'll post some comments there

Comment:  https://issues.redhat.com/browse/RHDEVDOCS-2384?focusedCommentId=15556330&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15556330
